components/constellation/constellation.rs

```diff
@@ -2800,6 +2804,10 @@ where
         debug!("Exiting GLPlayer thread.");
         WindowGLContext::get().exit();
 
+        // Wait for the canvas thread to exit before shutting down the font service, as
+        // canvas might still be using the system font service before shutting down.
+        let _ = canvas_exit_receiver.recv();
```

Comment:
Should we check error here?

Reply:
This is happening in the process of shutdown, so an error here likely means that the canvas thread has panicked or deadlocked, in which case I think the right thing to do is to keep trying to shut down.